File: frontend/collectr/types/collection.ts
import { CollectionType } from "./CollectionType";

export interface CollectionDTO {
  id: number;
  userId: number;
  userNickname: string;
  name: string;
  description: string | null;
  isPublic: boolean;
  status: number | null;
  collectionType: CollectionType;
  createdAt: string;
  updatedAt: string;
  active: boolean;
}

export interface CreateCollectionRequest {
  name: string;
  description?: string;
  isPublic: boolean;
  status?: number;
  collectionType: CollectionType;
}

export interface UpdateCollectionRequest {
  name: string;
  description?: string;
  isPublic: boolean;
  status?: number;
  collectionType: CollectionType;
}

File: backend/src/main/java/com/lykawka/collectr/app/model/collection/Collection.java
package com.lykawka.collectr.app.model.collection;


import com.lykawka.collectr.app.model.BaseEntity;
import com.lykawka.collectr.app.model.user.User;

import jakarta.persistence.*;
import lombok.*;

@Data
@EqualsAndHashCode(callSuper=false)
@Entity
@Table(name = "collections")
@NamedEntityGraph(name = "Collection.user", attributeNodes = @NamedAttributeNode("user"))
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Collection extends BaseEntity {
    @Column(nullable = false)
    private String name;

    @Column(length = 1000)
    private String description;

    @Column(nullable = false)
    private Boolean isPublic;

    @Column(nullable = false)
    private Integer status;

    @Enumerated(EnumType.STRING)
    private CollectionType collectionType;

    @Column(nullable = false)
    private Long userId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "userId", insertable = false, updatable = false)
    private User user;
}
